synapse/handlers/presence.py

```diff
@@ -490,7 +502,10 @@ async def user_syncing(
             # what the spec wants: see comment in the BasePresenceHandler version
             # of this function.
             await self.set_state(
-                UserID.from_string(user_id), {"presence": presence_state}, True
+                UserID.from_string(user_id),
+                device_id,
+                {"presence": presence_state},
+                True,
```

Comment:
Sorry, while we're here can we do these as keyword arguments please?

Reply:
(Below for the similar place in the `PresenceHandler` version too please)